docs/api/api.yml

```diff
@@ -0,0 +1,594 @@
+  version: "0.0.1"
+  title: Sidetree REST API
+tags:
+  - name: Sidetree
+    description: APIs for interacting with the Sidetree Node.
+  - name: Blockchain
+    description: APIs for interacting with the Blockchain system.
+  - name: Content Addressed Storage
+    description: APIs for interacting with the CAS system.
+paths:
+  /sidetree/operations:
+    post:
+      tags:
+        - Sidetree
+      summary: Writes an Operation to Sidetree.
+      operationId: writeSidetreeOperation
+      parameters:
+        - in: body
+          schema:
+            oneOf:
+              - $ref: '#/components/schemas/SidetreeCreateOperation'
+              - $ref: '#/components/schemas/SidetreeUpdateOperation'
+              - $ref: '#/components/schemas/SidetreeRecoverOperation'
+              - $ref: '#/components/schemas/SidetreeDeactivateOperation'
+      responses:
+        '200':
+          description: Success
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+        '500':
+          description: Server Error
+  /identifiers/{did-with-or-without-initial-state}:
+    get:
+      tags:
+        - Sidetree
+      summary: Resolve a Sidetree DID.
+      operationId: resolveSidetreeDID
+      parameters:
+        - in: path
+          required: true
+          name: did-with-or-without-initial-state
+          schema:
+            type: string
+          description: A DID to resolve
+          example: "did:sidetree:EiAC2jrPmjaLI4xMiHTGWaKK29HmU9USFWA22lYc6CV0Bg?-sidetree-initial-state=eyJ0eXBlIjoiY3JlYXRlIiwic3VmZml4RGF0YSI6ImV5SnZjR1Z5WVhScGIyNUVZWFJoU0dGemFDSTZJa1ZwUW13Mk9IUktSRFp3YmxadVdHWTFURUZqY1VGWWJsRkhOR2syY2xKSGVuUmZlazEzYXpkaFZWUTBlVUVpTENKeVpXTnZkbVZ5ZVV0bGVTSTZleUp3ZFdKc2FXTkxaWGxJWlhnaU9pSXdNalE0WkRWaFlUbGxZamxqWVdZNE5EWmhNalZoTkRReE1qbGlPR013TURBek9HUTFObVJsTlROaVptTTNZbUU1TkRneU1tRTFNV1ZpTUdabU1EazNNbU1pZlN3aWJtVjRkRkpsWTI5MlpYSjVUM1J3U0dGemFDSTZJa1ZwUTI5aU5YZFZkMEV5U0VObVRGUjZjbmRHZG14b2JVSm5TRnB0ZEVsZmRXVXhNa1JuWHpsVlkxOXdlR2NpZlEiLCJvcGVyYXRpb25EYXRhIjoiZXlKdVpYaDBWWEJrWVhSbFQzUndTR0Z6YUNJNklrVnBSSEZMVDJ0ZlRsZHVZMkZrT0RJelYySm9WVGwyZUVwcmQwVnVTVFZHUlVNeU0xbDViRE5rUlZnNWJtY2lMQ0prYjJOMWJXVnVkQ0k2ZXlKQVkyOXVkR1Y0ZENJNkltaDBkSEJ6T2k4dmR6TnBaQzV2Y21jdlpHbGtMM1l4SWl3aWNIVmliR2xqUzJWNUlqcGJleUpwWkNJNklpTnphV2R1YVc1blMyVjVJaXdpZEhsd1pTSTZJbE5sWTNBeU5UWnJNVlpsY21sbWFXTmhkR2x2Ymt0bGVUSXdNVGdpTENKMWMyRm5aU0k2SW5OcFoyNXBibWNpTENKd2RXSnNhV05MWlhsSVpYZ2lPaUl3TTJKa01HVTBOREF3TlRKaU9UUXlaVE13T0dJNVptUXdPR1JpTWpsaFltTTRaRFl6TmpZNE5ESXpNMkZsT0Raa09Ea3lZVEk1WmpCak5qRTJabVV3TldVaWZWMHNJbk5sY25acFkyVWlPbHQ3SW1sa0lqb2lTV1JsYm5ScGRIbElkV0lpTENKMGVYQmxJam9pU1dSbGJuUnBkSGxJZFdJaUxDSnpaWEoyYVdObFJXNWtjRzlwYm5RaU9uc2lRR052Ym5SbGVIUWlPaUp6WTJobGJXRXVhV1JsYm5ScGRIa3VabTkxYm1SaGRHbHZiaTlvZFdJaUxDSkFkSGx3WlNJNklsVnpaWEpUWlhKMmFXTmxSVzVrY0c5cGJuUWlMQ0pwYm5OMFlXNWpaWE1pT2xzaVpHbGtPbk5wWkdWMGNtVmxPblpoYkhWbE1DSmRmWDFkZlgwIn0"
```

Comment:
decoding this results in  {"type":"create","suffixData"...  which is not our latest request

Reply:
@sandrask could you please provide an updated example?

Reply:
eyJkZWx0YV9oYXNoIjoiRWlCWE9wa2dGa1JuY0xETm9XQVdQT2lwSXBQMWxHS3Y4cmF5UFhCbWd1eC0yUSIsInJlY292ZXJ5X2NvbW1pdG1lbnQiOiJFaUNfTS1JSVAtMEs4V0dDeUlmbThZNjRZN1FwZjJ3U0x0cVN3QWtfSFlJRDh3IiwicmVjb3Zlcnlfa2V5Ijp7ImNydiI6IlAtMjU2Iiwia3R5IjoiRUMiLCJ4IjoiTmVKQ3lESU43NmZ6Z2NMMy0xRjBoOEFaa0tseDBqcmMwOXdOUTh0cUw5YyIsInkiOiJIWHpickRGRVNLT2tjcVJLbjhpamVkdDRRZEZ1bnhSUm1pM3ZGYmVrS093In19.eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJhZGQtcHVibGljLWtleXMiLCJwdWJsaWNfa2V5cyI6W3siaWQiOiJrZXkxIiwiandrIjp7ImNydiI6IlAtMjU2Iiwia3R5IjoiRUMiLCJ4IjoibUp4MFZPWnFLUDJ6Q3lRSno4OVJXRGxNTmc4X0djVnJBRlNnTUE4VHU3ayIsInkiOiJmbVFwcFZNa08tVnpJaGZTaVJBZjg4VkhQaTVqaHN4UEdtZ0I4amZGUGxVIn0sInR5cGUiOiJKd3NWZXJpZmljYXRpb25LZXkyMDIwIiwidXNhZ2UiOlsib3BzIl19LHsiaWQiOiJkdWFsLWF1dGgtZ2VuIiwiandrIjp7ImNydiI6IlAtMjU2Iiwia3R5IjoiRUMiLCJ4IjoiUGFwN1gyUEhWWW5tMFpTMEEtbWZaWV9tLVRwNmUzbEQ1TTJXMFBud0hUcyIsInkiOiJjVmt3ZlBDV3g5OFZ3UjBLcmJnUTFPTzNrQ3BFcGFMRTV4aGo0QVlzRXRjIn0sInR5cGUiOiJKd3NWZXJpZmljYXRpb25LZXkyMDIwIiwidXNhZ2UiOlsiYXV0aCIsImdlbmVyYWwiXX0seyJpZCI6ImR1YWwtYXNzZXJ0aW9uLWdlbiIsImp3ayI6eyJjcnYiOiJFZDI1NTE5Iiwia3R5IjoiT0tQIiwieCI6IlBJaWFyX2o4ZW5HMkEyTXVhN3I0MW5DbTJIOHFtY25Kd25Va0tpdGxYWlkiLCJ5IjoiIn0sInR5cGUiOiJFZDI1NTE5VmVyaWZpY2F0aW9uS2V5MjAxOCIsInVzYWdlIjpbImFzc2VydGlvbiIsImdlbmVyYWwiXX1dfSx7ImFjdGlvbiI6ImFkZC1zZXJ2aWNlLWVuZHBvaW50cyIsInNlcnZpY2VfZW5kcG9pbnRzIjpbeyJpZCI6Im9pZGMiLCJzZXJ2aWNlRW5kcG9pbnQiOiJodHRwczovL29wZW5pZC5leGFtcGxlLmNvbS8iLCJ0eXBlIjoiT3BlbklkQ29ubmVjdFZlcnNpb24xLjBTZXJ2aWNlIn0seyJpZCI6Imh1YiIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHBzOi8vaHViLmV4YW1wbGUuY29tLy5pZGVudGl0eS9kaWQ6ZXhhbXBsZTowMTIzNDU2Nzg5YWJjZGVmLyIsInR5cGUiOiJIdWJTZXJ2aWNlIn1dfV0sInVwZGF0ZV9jb21taXRtZW50IjoiRWlDT3hBbEZ6Rjd4N0ZQZGRaQ1ZYaHJjZDJEZ2RGeVZJS1NKMm0xT21FZ0JSZyJ9

Reply:
@troyronda see above

Reply:
will add examples in a followup.